File: mdds_dto/task_dto.py
# Copyright (c) 2025 Oleksiy Oleksandrovych Sayankin. All Rights Reserved.
# Refer to the LICENSE file in the root directory for full license details.

"""Data Transfer Object for Task"""

from pydantic import BaseModel
from typing import List
from datetime import datetime


class TaskDTO(BaseModel):
    task_id: str
    date_time_created: datetime
    matrix: List[List[float]]
    rhs: List[float]
    slae_solving_method: str

File: src/main/java/dto/TaskDTO.java
/*
 * Copyright (c) 2025 Oleksiy Oleksandrovych Sayankin. All Rights Reserved.
 * Refer to the LICENSE file in the root directory for full license details.
 */

package dto;

import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/** Represents Data Transfer Object for the task for Executor. */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class TaskDTO {
  private String id;
  private Instant dateTime;
  private double[][] matrix;
  private double[] rhs;
  private SlaeSolver slaeSolvingMethod;
}
